### ISSUE TITLE
TOP priority for Ailey

### ISSUE THREAD

**Issue by gkiar** [NeuroDataDesign/seelviz-f16s17#53]
**A registered brain to Allen atlas ingested into ndstore, with ndviz link showing the brain overlaid with the Atlas.**

This means you need to, from square one:
- [x] download her CLARITY brain from ndstore at resolution 5 using ndreg/ndio
- [x] align it to the ARA using ndreg
- [ ] upload the atlas and brain to a token in ndstore using ndreg/ndio
- [ ] provide an ndviz link to the project with the two channels shown

**Comment by toekneesunshine** [comment]
1. Aut1367 raw brain ingested into ndstore and viewable on [ndviz](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_raw_8/5/328/443/250#) saved as token Aut1367_seelviz and channel Aut1367_raw_8.
2. Aut1367 atlas aligned brain into a token Aut1367_seelviz and viewable on [ndviz](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_annotated_LPS_8/5/328/443/250) as channel Aut1367_annotated_LPS_8.
3. ndviz [link](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_raw_8,Aut1367_annotated_LPS_8/5/328/443/250#) with two channels overlaid in native brain orientation (LPS).  In order to view overlay, change opacity on the Aut1367_annotated_LPS_8 channel such that the opacity isn't 100% (do around 50%).  TODO:  Figure out how to upload the Matplotlib colored annotated version...  See example overlay [here](https://github.com/alee156/cldev/blob/master/Tony/images/nov28/exampleoverlay.png).

**Comment by gkiar** [comment]
@toekneesunshine what's up with the [brain](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_raw_8/5/328/443/250#)? The colour seems inverted or something? I think your implementation histogram equalization is going wrong somewhere.

[This](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_raw_8,Aut1367_annotated_LPS_8/5/328/443/250#) image is getting close but isn't what we want. We want a **single** token containing **two channels** (which you do). We want the types of the channels, however to be: an **image channel** containing the aligned brain, and an **annotation channel** containing the ARA (currently the ARA is also an image channel). With the ARA as an annotation channel it'll automatically colour different regions.

Once we fix the histogram eq. and have the ARA as an annotation, it's looking close!!

**Comment by toekneesunshine** [comment]
Hey @gkiar, I was actually going to ask that about the brain today -- Albert and I thought the brain looked very odd.  Not sure what's going on with the raw image, that's the brain image we're getting directly from ndstore w/o any transformations.  To confirm that, I actually uploaded the brain twice -- [here](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_rawest_8/5/297/481/250) is the raw image I get directly from ndstore (by directly downloading the img from ndstore, and it looks identical to the raw image I uploaded previously...)

As for uploading the annotation channel as an image channel, let me try that now -- didn't realize there was an annotation image type...!

**Comment by gkiar** [comment]
Looking at Ailey's data in ndviz at the same location I'm inclined to disagree with you :) Where are you pulling it from?

http://viz.neurodata.io/Aut1367/Aut1367_stitched/5/297/481/250

**Comment by mlee156** [comment]
We're pulling from ndstore using ndreg's function call:

`inToken = "Aut1367"`

`inImg = imgDownload(inToken, resolution=5)`

`imgUpload(inImg, token, channel, resolution=5)`

where the token and channel are the ones that we sent you

**Comment by toekneesunshine** [comment]
Yeah, what Albert said, except instead of inToken = "Aut1374" we're using inToken = "Aut1367".

**Comment by gkiar** [comment]
Ok, verify the following:
  - the datatype you're uploading as is the same as the original data -- if you are doing a different datatype without scaling appropriately you'll have overflow error looking like this.
  - you're in fact doing what you just told me you're doing, otherwise.

If these are the case, comment back here documenting clearly that you:

1. viewed the brain using ndviz (provide the link you used to view it initially that looks ok, like the link I posted above)
2. pulled using a certain url (provide the specific url that ndreg submits, not the ndreg function call)
3. pushed the data back to a different channel immediately (provide the url that ndreg posts, not the ndreg function call)
4. viewed the brain using ndviz and got something nonsensical (provide link to ndviz of the uploaded brain)

**Comment by gkiar** [comment]
If you do this and I can't find an obvious problem, we'll copy-pasta it to an issue in ndstore

**Comment by toekneesunshine** [comment]
Hey @gkiar:
By uploading as an annotation channel, I was able to get [color annotations](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_annotated_LPS_8_PROPER/5/320/508/250).  However, the annotations channel still seems to be cut off... not sure how to fix this problem currently.

So the current overlay looks something like [this](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_raw_8,Aut1367_annotated_LPS_8_PROPER/5/320/508/250#).

TODO:
1.  Figure out why exactly the annotated image is cut off (the image on my ipynb is fine, so it's probably some upload configuration).
2.  Try to find out if the raw data we're using is appropriate.

Is there a docs page that shows the specific url ndreg submits when me make a particular function call?  The only ndreg docs I could find were Kwame's ndreg [ipynb's](https://github.com/neurodata/ndreg/tree/master/doc).

**Comment by gkiar** [comment]
Your urls aren't working for me?

The only way I can think of is going into the ndreg code and adding a `print` statement in the function call that you're using that prints the url. If it's using ndio, then check the ndio docs to see if there is an additional parameter you can pass so that you see the url, such as a `verbose` flag.

**Comment by toekneesunshine** [comment]
Hey @gkiar, you must've caught them just as I was in the middle of reuploading something!  The links are working now.

As a temporary workaround, I was able to get the annotated labels overlaid on the original Aut1367 token.  There's a picture of that [here](https://github.com/alee156/cldev/blob/master/Tony/images/nov28/overlayonAut1367.png).  I did this manually by using the (+) icon on the side to add an additional channel overlay -- if I can figure out why it's cut off, this looks pretty close to what Ailey's looking for, right?

**Comment by gkiar** [comment]
if your brain were histogrammed/datatyped correctly and solved the cutoff so looked like the thing you showed me in the picture, then the link you provided in "the current overlay looks something like this" would be perfect. Let's figure out why the brain looks dumb and why it's cutoff, then we're there!

**Comment by toekneesunshine** [comment]
Hmm, it also appears that some of the layers didn't get cut off...
Here's an [example of one that wasn't cut off](http://viz.neurodata.io/Aut1367_seelviz/Aut1367_raw_8,Aut1367_annotated_LPS_8_PROPER/5/320/508/230#).

Not sure what's going on, I guess we can figure it out more during class/for next week.

**Comment by gkiar** [comment]
remind me to ask you to show the "cut off" slices to me today in class.

**Comment by toekneesunshine** [comment]
So as it turns out, after trying a bunch of different layers, ONLY later 250 is cut off.  I'm actually pretty sure that might be b/c when I was uploading, I was also refreshing layer 250 (I had the page open) so something funny happened.  I think if I reupload the image it might fix it -- I'll do that after class.  Here's an example overlay over the [native Aut1367 token](https://github.com/alee156/cldev/blob/master/Tony/images/nov28/overlayonAut1367PROPER.png) (by using the (+) icon again).

**Comment by gkiar** [comment]
Interesting. Try reuploading only slice 250 and see if that fixes it.
